### PR TITLE
Add class_name setting, update Java generation

### PR DESCRIPTION
**Pull request by michaelbausor**
Mass changes due to signature change of getApiWrapperClassName to accept InterfaceConfig instead of Interface

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/library_gapic.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@ language_settings:
   java:
     package_name: com.google.gcloud.pubsub.spi
     release_level: BETA
+    class_names:
+      google.example.library.v1.LibraryService: LibraryClient
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/library_gapic.yaml]: The settings class isn't getting renamed. The two of them need to be in sync.

I suggest a different approach where the base name is given. Something like:

```
  interface_names:
    google.example.library.v1.LibraryService: Library
```

And this generates `LibraryClient` and `LibrarySettings` for classes, and `libraryClient` and `librarySettings` for variable names.

Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/library_gapic.yaml]: What is the major use case of this field? Is it only for renaming interface names or any class names?

Comment (michaelbausor) [inline src/test/java/com/google/api/codegen/testdata/library_gapic.yaml]: It is for renaming from LibraryServiceClient->LibraryClient, and all other appropriate classes/variables.

I had hoped to allow for overriding the standard Client suffix as well, but you are right that this change must apply to the settings classes, and test class. Have updated accordingly.

**Comment by michaelbausor** [comment]
PTAL

File: src/test/java/com/google/api/codegen/testdata/library_gapic.yaml (hunk 1 above)
Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/library_gapic.yaml]: Calling it interface_names indicate it only overrides class names for interfaces. I would prefer to keep the field name as "class_names"

Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/library_gapic.yaml]: That's what I wanted! I don't think we should use this for other renames. This rename applies to things besides classes anyway (variable names).

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**